### PR TITLE
Add `embed` parameter to framework dependency

### PR DESCRIPTION
**Pull request by zdnk**
Resolves https://github.com/tuist/tuist/issues/454

I basically just followed the code and was fixing issues.

I am unable to run tests as my Xcode crashes every time I try, but they compile.
Iam not sure if the implementation is correct, I am not sure how to properly test it, but I hope it will help a little and someone can finish it :)

I also tried to make it non-breaking change.

### CODE REVIEW

File: Sources/ProjectDescription/TargetDependency.swift [hunk 1]
```diff
@@ -30,7 +30,8 @@ public enum TargetDependency: Codable, Equatable {
     ///
     /// - Parameters:
     ///   - path: Relative path to the prebuilt framework
-    case framework(path: String)
+    ///   - embed: Should be this framework embedded into aplication
```
Comment (ollieatkinson) [inline Sources/ProjectDescription/TargetDependency.swift]: This sentence doesn't read well

File: Sources/ProjectDescription/TargetDependency.swift [hunk 2]
```diff
@@ -112,7 +124,10 @@ extension TargetDependency {
             )
 
         case "framework":
-            self = .framework(path: try container.decode(String.self, forKey: .path))
+            self = .framework(
+                path: try container.decode(String.self, forKey: .path),
+                embed: try container.decodeIfPresent(Bool.self, forKey: .embed) ?? true
```
Comment (ollieatkinson) [inline Sources/ProjectDescription/TargetDependency.swift]: Thanks for thinking about backwards compatibility here :+1:

**Comment by kwridan** [comment]
Hi @zdnk, 

Digging further into this, looks like there was a bug in the logic that was determining if precompiled frameworks needed to be embed or not. 

Can you test out https://github.com/tuist/tuist/pull/490 on your project? If it works, I believe it would alleviate the need to explicitly need an `embed` flag.

Thanks!

**Comment by pepibumur** [comment]
We can close this one considering we were able to reproduce the issue and the fix has already been merge. If the issue we reproduced turns out not to be the same we can reopen the PR.
Thanks @kwridan & @zdnk 👏

**Comment by zdnk** [comment]
Well, Xcode 11 has option to select if it should embed, embed & sign or do nothing with the linked frameworks, this should maybe be also in tuist for whatever reason Xcode is doing that?

**Comment by pepibumur** [comment]
> Well, Xcode 11 has option to select if it should embed, embed & sign or do nothing with the linked frameworks, this should maybe be also in tuist for whatever reason Xcode is doing that?

You are right. The approach Tuist takes is slightly different. Rather than you making the decision, which might be error prone, we let Tuist make the decision decision for you. Knowing what to embed/link is straightforward with a few frameworks, but things get more complicated when the dependencies graph grows.

The case presented in these issue was clearly a bug that has been addressed. If there are other issues that prevent you from linking your frameworks/libraries, we are happy to discuss them.

**Pull request closed without merging**